EngineLayer/CrosslinkSearch/CrosslinkSearchEngine.cs

```diff
@@ -128,8 +128,18 @@ protected override MetaMorpheusEngineResults RunSpecific()
                     {
                         if (CrosslinkSearchTopN)
                         {
-                            // take top N hits for this scan
-                            idsOfPeptidesPossiblyObserved = idsOfPeptidesPossiblyObserved.OrderByDescending(p => scoringTable[p]).Take(TopN).ToList();
+                            //// take top N hits for this scan
+                            //var scoreCutList = idsOfPeptidesPossiblyObserved.OrderByDescending(p => scoringTable[p]).ThenBy(p => PeptideIndex[p].FullSequence).ThenBy(p => PeptideIndex[p].Protein.Accession).ToList();
+                            //if (scoreCutList.Count >= TopN)
+                            //{
+                            //    var scoreCut = scoreCutList[TopN - 1];
+                            //    idsOfPeptidesPossiblyObserved = scoreCutList.Where(p => p > scoreCut).ToList();
+                            //}
+                            //else
+                            //{
+                            //    idsOfPeptidesPossiblyObserved = scoreCutList;
+                            //}
+                            idsOfPeptidesPossiblyObserved = idsOfPeptidesPossiblyObserved.OrderByDescending(p => scoringTable[p]).ThenBy(p => PeptideIndex[p].FullSequence).ThenBy(p => PeptideIndex[p].Protein.Accession).Take(TopN).ToList();
```

Comment:
For some reasons, it seems like the xl search slows down, I think it is due to the ordering. Need more tests.

Reply:
This change slows the search ~10 times on one dataset. I tried a few ways but none is good enough.

Reply:
This has been changed to a new in place sort. Let's see if the search times are more reasonable now

Reply:
1. Original no ordering
Time to run task: 01:58:27.0898884
2. OrderBy.ThenBy
Time to run task: 15:44:23.8345173
3. Sort 
Time to run task: 12:34:51.5948920
Sort is better than OrderBy!

Reply:
not much better unfortunately